go-controller/pkg/ovn/master.go

```diff
@@ -1491,3 +1493,177 @@ func (oc *Controller) syncNodesRetriable(nodes []interface{}) error {
 	}
 	return nil
 }
+
+type nodeSyncs struct {
+	syncNode          bool
+	syncClusterRouter bool
+	syncMgmtPort      bool
+	syncGw            bool
+}
+
+func (oc *Controller) addUpdateNodeEvent(node *kapi.Node, nSyncs *nodeSyncs) error {
+	var hostSubnets []*net.IPNet
+	var errs []error
+	var err error
+
+	if noHostSubnet := noHostSubnet(node); noHostSubnet {
+		err := oc.lsManager.AddNoHostSubnetNode(node.Name)
+		if err != nil {
+			return err
+		}
+	}
+
+	klog.Infof("Adding Node %q", node.Name)
+	if nSyncs.syncNode {
+		if hostSubnets, err = oc.addNode(node); err != nil {
+			oc.addNodeFailed.Store(node.Name, true)
+			oc.nodeClusterRouterPortFailed.Store(node.Name, true)
+			oc.mgmtPortFailed.Store(node.Name, true)
+			oc.gatewaysFailed.Store(node.Name, true)
+			return fmt.Errorf("nodeAdd: error creating subnet for node %s: %w", node.Name, err)
+		}
+		oc.addNodeFailed.Delete(node.Name)
+	}
+
+	if nSyncs.syncClusterRouter {
+		if err = oc.syncNodeClusterRouterPort(node, nil); err != nil {
+			errs = append(errs, err)
+			oc.nodeClusterRouterPortFailed.Store(node.Name, true)
+		} else {
+			oc.nodeClusterRouterPortFailed.Delete(node.Name)
+		}
+	}
+
+	if nSyncs.syncMgmtPort {
+		err := oc.syncNodeManagementPort(node, hostSubnets)
+		if err != nil {
+			errs = append(errs, err)
+			oc.mgmtPortFailed.Store(node.Name, true)
+		} else {
+			oc.mgmtPortFailed.Delete(node.Name)
+		}
+	}
+
+	// delete stale chassis in SBDB if any
+	oc.deleteStaleNodeChassis(node)
```

Comment:
what do u mean the retry code will retry this as its not controller by a flag ?

Reply:
I think she means it doesn't return an error type, so how do we know if it was successful or not. It's a good point, right now that function just logs an error, perhaps it should return an error and we propagate that up so it will get retried.